tests/test_merge_memory_contract.py

```
@@ -0,0 +1,155 @@
+import sys
+from pathlib import Path
+
+REPO_ROOT = Path(__file__).resolve().parents[1]
+SCRIPT = REPO_ROOT / "scripts" / "merge_memory_contract.py"
+
+MARKER_START = "<!-- memory-contract:start -->"
+MARKER_END = "<!-- memory-contract:end -->"
+
+
+def _run(args: list[str]) -> subprocess.CompletedProcess[str]:
+    return subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)